contrib/coredns/policy/policy_test.go

```diff
@@ -0,0 +1,105 @@
+package policy
+
+import (
+	"testing"
+	"errors"
+	"github.com/miekg/dns"
+	"net"
+
+)
+
+
+
+// This function tests that Validation of Middleware response is working properly
+// Inputs are t *testing.T
+// Output is the error update in t
+func TestValidateMiddlewareResponse(t *testing.T) {
+	var p PolicyMiddleware
+	var dnsResponseWriter	dns.ResponseWriter
+
+	dnsMsg := new(dns.Msg)
+	//
+	// Case 1
+	//
+	var response1 Response
+
+	// Case 1a - Err is set, return should have err and RcodeServerFailure
+	err := errors.New("Invalid error")
+	status := dns.RcodeServerFailure
+	returnedStatus, returnedError := ValidateMiddlewareResponse(&p, dnsResponseWriter, dnsMsg, &response1, status, err)
```

Comment:
We need unit tests for the whole ServeDNS method and other methods. This is probably a stretch for this PR.

I am thinking is that you will make a pep "Client" interface, and change this struct:

https://github.com/infobloxopen/policy-box/blob/master/pep/client.go#L25

to a "client" (lowercase) that implements the Client interface.

Then, in maybe a "test" directory you will add another implementation "testClient", which also has a "SetResponse" method. Then you'll need to set the "pdp" in the policy middleware to this test client. Then, you can make calls to SetResponse to tell the testClient what to reply with just before the test, then call through the ServeDNS method to test all the logic. You'll also need to do similar with the "Next" middleware.

All this can wait though, don't spend a ton of time on it now.